lib/agent_runner.rb

```diff
@@ -58,9 +58,14 @@ def self.register(agent)
   end
 
   def self.with_connection
-    ActiveRecord::Base.connection_pool.with_connection do
-      yield
-    end
+    begin
+  	  ActiveRecord::Base.connection_pool.with_connection do
+  			  yield
+  	  end
+  	  rescue ActiveRecord::StatementInvalid => e
```

Comment:
Probably StatementInvalid is too wide and maybe we should catch only PG::ConnectionBad